File: example/directory/directory_android.go
// +build android

package main

import (
	"github.com/inkeliz/gojni"
)

//go:generate javac -source 8 -target 8 -bootclasspath $ANDROID_HOME\platforms\android-29\android.jar -classpath ..\..\register_android.jar -d classes directory_android.java
//go:generate jar cf directory_android.jar -C classes .

func pathDirectory(c Environment) (path string, err error) {
	sess, err := jni.NewSessionRegistered("com/inkeliz/example/directory")
	if err != nil {
		return path, err
	}

	function := androidEnvironment(c)

 	// If function is "Download" it will call "com/inkeliz/example/directory . Download()":
	return jni.NewCall(sess, function).ReturnString(nil)
}

func androidEnvironment(c Environment) string {
	switch c {
	case Download:
		return "Download"
	case Pictures:
		return "Pictures"
	case Music:
		return "Music"
	case Video:
		return "Video"
	case Documents:
		return "Document"
	default:
		return "Document"
	}
}

File: example/directory/directory_android.java
package com.inkeliz.example.directory;

import java.lang.Runnable;
import java.lang.String;
import java.io.File;
import android.os.Environment;
import android.content.Context;
import android.content.ClipboardManager;
import android.content.ClipData;
import android.content.Intent;
import android.util.Log;
import android.app.Activity;
import android.app.Fragment;
import android.app.FragmentManager;
import android.app.FragmentTransaction;
import android.app.PendingIntent;
import android.content.IntentSender;
import android.content.IntentSender.OnFinished;
import android.content.IntentSender.SendIntentException;
import android.os.Bundle;
import android.Manifest;
import android.content.pm.PackageManager;

// That import is used to call GOJNI
import github.com.inkeliz.gojni.register_android;

public class directory_android extends Fragment {
	Context ctx;
	final int PERMISSIONS_REQUEST = 1;

	public directory_android() {
	}

	@Override public void onAttach(Context ctx) {
		super.onAttach(ctx);
		this.ctx = ctx;

        AskPermission();

        // That is used to call GOJNI and make directory_android accessible from Golang
        register_android.Register(this);
	}

	public String Pictures() {
	    AskPermission();
        return Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DCIM).getAbsolutePath();
    }

	public String Download() {
	    AskPermission();
        return Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).getAbsolutePath();
    }

	public String Music() {
	    AskPermission();
        return Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_MUSIC).getAbsolutePath();
    }

	public String Video() {
	    AskPermission();
        return Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_MOVIES).getAbsolutePath();
    }

	public String Document() {
	    AskPermission();
        return Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS).getAbsolutePath();
    }

    public void AskPermission() {
        if (ctx.checkSelfPermission(Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED || ctx.checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
                requestPermissions(new String[]{Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE}, PERMISSIONS_REQUEST);
        }
    }
}